src/formatter/json_formatter.js

```diff
@@ -156,7 +161,7 @@ export default class JsonFormatter extends Formatter {
         data.result.duration = testStep.result.duration
       }
       if (status === Status.FAILED && exception) {
-        data.result.error_message = exception.stack || exception
+        data.result.error_message = stringifyException(exception)
```

Comment:
Thanks for the great test cases but thinking more on this feature I think we should use [node-assertion-error-formatter](https://github.com/charlierudolph/node-assertion-error-formatter) which is what we use when displaying errors in the terminal. It includes the logic for concatenating the message / stack and shows diffs as well. Sorry I didn't remember this sooner.

Reply:
node-assertion-error-formatter does not pass 3 of the tests case I wrote. If those error are acceptable I can modify for the fourth time the pull request :)

 1) ErrorHelpers                                                                                     
      stringifyException                                                                             
        return stack:                                                                                                                                                                            
     AssertionError: expected '[object Object]stack' to equal 'stack'                                
                                                                                                     
 2) ErrorHelpers                                                                                     
      stringifyException                                                                             
        return message and stack:                                                                                                                                                   
     AssertionError: expected 'errorstack' to equal 'error\nstack'                                   
                                                                                                     
 3) ErrorHelpers                                                                                     
      stringifyException                                                                             
        stringify an object:                                                                         
    TypeError: stack.indexOf is not a function

Reply:
I just released v2.0.1 of `node-assertion-error-formatter` which should fix cases 2 and 3 (https://github.com/charlierudolph/node-assertion-error-formatter/commit/ca21cfd552ce090b0273bd160e44b7cb54f7c071). Please update the version in the package.json / yarn.lock. 

I think test case number 1 needs to be updated so the message is an empty string instead of undefined. Testing out the error constructor in node 8, this was always the case.

If all your test cases pass, I think the tests / util file you added can be deleted as the test cases are now covered in `node-assertion-error-formatter`. We can use `node-assertion-error-formatter` directly in `src/formatter/json_formatter.js`